src/meta/src/manager/catalog/mod.rs

```diff
@@ -1984,9 +1981,7 @@ impl CatalogManager {
         let table_key = (table.database_id, table.schema_id, table.name.clone());
         assert!(
             !database_core.sources.contains_key(&source.id)
-                && !database_core.tables.contains_key(&table.id)
-                && database_core.has_in_progress_creation(&source_key)
-                && database_core.has_in_progress_creation(&table_key),
+                && !database_core.tables.contains_key(&table.id),
```

Comment:
Could you please shed some light on this removal?

Reply:
When `clean_dirty_tables`, we will clean up all the in memory tracking state as well.

As a result, these assertions could fail.

Reply:
Reason being, we could have the following sequence of events:
1. Clean dirty tables, causing the creation trackers to be cleared.
2. The running stream job fails, triggering cleanup via this function (or other function, depending on stream job type).
3. We hit the assertion, but since it could be already cleared in 1. so we do not need to check for it.